### ISSUE TITLE
Where is the ability to add pixel-wise padding?

### ISSUE THREAD

**Issue by lababidi** [keras-team/keras#8140]
I'm attempting to pad all pixels in an image as seen below to perform a proper Conv2DTranspose with padding...

```
[
[1, 4], 
[7, 9]
] 
-> 
[
[0,0,0,0,0], 
[0,1,0,4,0], 
[0,0,0,0,0],
[0,7,0,9,0],
[0,0,0,0,0],
]
```
But I can't seem to find any function in keras to do this. Any assistance is appreciated.

![padding_strides_transposed](https://user-images.githubusercontent.com/495399/31577402-a4c70f0a-b0db-11e7-9b38-7d92fbb8a3c5.gif)

**Comment by alxy** [comment]
This should be possible using tensorflow (I don't think there is a native keras solution). One implementation I can think of are ``SparseTensor``s. Considering the example you gave:

```
import tensorflow as tf

inputs = tf.SparseTensor(indices=[[1, 1], [3, 1], [1, 3], [3, 3]], values=[1, 4, 7, 9], dense_shape=[5, 5])
padded = tf.sparse_tensor_to_dense(inputs, validate_indices=False)
```
```
[[0 0 0 0 0]
 [0 1 0 7 0]
 [0 0 0 0 0]
 [0 4 0 9 0]
 [0 0 0 0 0]]
```

**Comment by lababidi** [comment]
@alxy Thanks for the suggestion, but that's assuming you know the size of the incoming tensor. In this case for a typical convolution, you do not know the size.

@fchollet any thoughts?
